pkg/endpoint/bpf.go

```diff
@@ -1161,8 +1161,18 @@ func (e *Endpoint) ApplyPolicyMapChanges(proxyWaitGroup *completion.WaitGroup) e
 		return err
 	}
 
-	// Ignoring the revertFunc; keep all successful changes even if some fail.
-	err, _ = e.updateNetworkPolicy(proxyWaitGroup)
+	// Only update Envoy if there are envoy redirects.
+	// This is safe to do here, since a PolicyMapChange cannot
+	// cause an envoy redirect to appear or disappear. It only allows for
+	// incremental updates. Thus, we don't need to worry about stale
+	// policy not being cleaned up.
+	if e.desiredPolicy.L4Policy.HasEnvoyRedirect() {
```

Comment:
add `|| e.isIngress` to the condition, as Ingress endpoint also needs to enforce L3/4 policy.

Reply:
should we add one more condition for proxylib `e.desiredPolicy.L4Policy.HasProxylibRedirect()` as well ?

Reply:
Proxylib is covered by Envoy check already, as it is also implemented in Envoy:
```
pkg/policy/l4.go:       redirectTypeProxylib redirectTypes = 1<<iota | redirectTypeEnvoy
```